lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3168,6 +3178,16 @@ void VisitNestedScopes(ParseNode* pnodeScopeList, ParseNode* pnodeParent, ByteCo
                 // Push the param scope
                 byteCodeGenerator->PushScope(paramScope);
 
+                /*
+                We need this call to happen *before* visiting the param scope. If we have a split scope,
+                and this function has parameters that default to a lambda which uses the implicit "arguments" inside
+                of eval, we need this function's argumentsSymbol to be set already, so that we can mark it as
+                non-locally referenced inside of PostVisitFunction.
+
+                See Github issue #1209 for more.
+                */
+                AddVarsToScope(pnodeScope->sxFnc.pnodeVars, byteCodeGenerator);
```

Comment:
Good question. I don't have a solid answer to that. If you're referring to the setting of `argumentsSymbol` inside of `AddVarsToScope`, my thinking would be that we set it there because that already has to do a pass over all of the variables in the body scope, and doing it elsewhere would mean doing two passes. In the more common case where we don't have a split scope, we don't do any param scope processing logic, so we can't only set it there. 

@ianwjhalliday thoughts?